src/App.tsx

```diff
@@ -146,10 +146,26 @@ export default function App() {
   const [errorsDialog, setErrorsDialog] = React.useState(false);
   const dispatch = useDispatch();
   const etebase = useCredentials();
-  const darkMode = useSelector((state: store.StoreState) => state.settings.darkMode);
+  const darkModeUserSelection = useSelector((state: store.StoreState) => state.settings.darkModeUserSelection);
   const fetchCount = useSelector((state: store.StoreState) => state.fetchCount);
   const errors = useSelector((state: store.StoreState) => state.errors);
 
+  function shouldBeDark(userSelection: string): boolean {
+    if (userSelection === "auto") {
+      return window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)").matches;
```

Comment:
```suggestion
      return window?.matchMedia("(prefers-color-scheme: dark)").matches;
```
I think that's the short syntax, though maybe it's `.?` please double check.